### PR TITLE
[Intel MKL] Enabling BFloat16 element-wise ops using MKLDNN API - Rou…

### PR DESCRIPTION
**Pull request by nhasabni**
…nd 2

This PR enables support for BFloat16 versions of element-wise arithmetic ops
and activation functions using MKL-DNN API.

### CODE REVIEW

**Review by shahzadlone — DISMISSED** [review]

**Comment by nhasabni** [comment]
Thanks @shahzadlone for approval! Would you mind taking a look at first part of this PR #27300?

File: tensorflow/core/ops/nn_ops.cc [hunk 1]
```diff
@@ -1999,7 +1999,7 @@ REGISTER_OP("_MklLeakyRelu")
     .Input("mkl_features: uint8")
     .Output("activations: T")
     .Output("mkl_activations: uint8")
-    .Attr("T: {half, float, double} = DT_FLOAT")
+    .Attr("T: realnumbertype = DT_FLOAT")
```
Comment (penpornk) [inline tensorflow/core/ops/nn_ops.cc]: `realnumbertype` supports integrals. Do you want to allow int as well?
https://github.com/tensorflow/tensorflow/blob/526f6337bbbaee4492156d358b9d5e90d1c87382/tensorflow/core/framework/op_def_builder.cc#L135-L140
https://github.com/tensorflow/tensorflow/blob/526f6337bbbaee4492156d358b9d5e90d1c87382/tensorflow/core/framework/types.h#L274-L279

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for your PR and sorry for my delay! I have one question.

File: tensorflow/core/ops/nn_ops.cc (hunk 1 above)
Comment (nhasabni) [reply tensorflow/core/ops/nn_ops.cc]: Good point. Fixed.

**Review by penpornk — APPROVED** [review]
Thank you for contributing!

**Comment by nhasabni** [comment]
Thanks for approval @penpornk!

**Pull request merged**